plugins/processors/starlark/starlark.go

```diff
@@ -40,6 +41,46 @@ type Starlark struct {
 	applyFunc *starlark.Function
 	args      starlark.Tuple
 	results   []telegraf.Metric
+	cache     Cache
+}
+
+// The struct that defines the shared cache used
+type Cache struct {
+	content map[string]interface{}
+	mutex   sync.RWMutex
+}
+
+// Load from the shared cache the value corresponding to the given key
+func (c *Cache) Load(key string) starlark.Value {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	var value = c.content[key]
+	if value == nil {
+		return starlark.None
+	}
+	switch v := value.(type) {
+	case telegraf.Metric:
+		return &Metric{metric: v}
```

Comment:
why do we do this? and does this have any implications for maps of metrics or slices of metrics?